src/huggingface_hub/inference/_common.py

```diff
@@ -186,9 +185,29 @@ def _open_as_binary(content: Optional[ContentT]) -> Generator[Optional[BinaryT],
         logger.debug(f"Opening content from {content}")
         with content.open("rb") as f:
             yield f
-    else:
-        # Otherwise: already a file-like object or None
-        yield content
+        return
+
+    # If content is a PIL Image => convert to bytes
+    if is_pillow_available():
+        from PIL import Image
+
+        if isinstance(content, Image.Image):
+            logger.debug("Converting PIL Image to bytes")
+            buffer = io.BytesIO()
+            content.save(buffer, format="PNG")
+            yield buffer.getvalue()
+            return
+
+    # Check if content is a PIL Image but PIL is not available
+    if TYPE_CHECKING:
+        # For type checking purposes, assert that content is not an Image at this point
+        assert not hasattr(content, "save"), "Image objects should be handled above"
+    elif hasattr(content, "save") and hasattr(content, "format"):
+        # Likely a PIL Image but PIL not available - this is an error condition
+        raise ImportError("PIL Image detected but Pillow is not available. Please install Pillow.")
```

Comment:
This is not needed. If it's "just" for type hints, better to put a `# type: ignore` in this case.

```suggestion
```

Reply:
>        # Likely a PIL Image but PIL not available - this is an error condition

This is just not possible (since if a PIL.Image is passed, then means PIL is installed)